### PR TITLE
add terraform test cases about volume and keypair

### PR DESCRIPTION
**Pull request by lanfon72**
## Changes
- **UPDATE** `terraform_install.sh` to fix download latest version bug
- **UPDATE** `.gitignore` to exclude all folders in `terraform_test_artifacts/`
- **NEW** configure option `terraform-provider-harvester` to configure version of **Harvester Terraform Provider**
- **NEW** binary tool `json2hcl` to convert json format into hcl (terraform) format
- **NEW** fixture `fixtures/terraform.py` for terraform related test cases
- **NEW** Test cases
    1. `test_create_ssh_key` for legacy `test_create_keypairs_using_terraform`
    2. `test_delete_ssh_key`
    3. `test_create_volume` for legacy `test_create_volume_using_terraform`
    4. `test_delete_volume`

### CODE REVIEW

**Comment by lanfon72** [comment]
Tested on `harvester-runtests#407`
The filename `test_z` is necessary, we would need to let terraform related test cases be executed in the last.

File: harvester_e2e_tests/fixtures/terraform.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            return out
+
+        def ssh_key(self, resource_name, name, public_key, *, convert=True, **kwargs):
+            spec = {"name": name, "public_key": public_key}
+            spec.update(kwargs)
+            rv = dict(resource=dict(harvester_ssh_key={resource_name: spec}))
+
+            if convert:
+                return ResourceContext("harvester_ssh_key", resource_name, self.convert_to_hcl(rv))
+            return rv
```
Comment (albinsun) [inline harvester_e2e_tests/fixtures/terraform.py:148]: May extract `resource_type`.
```
resource_type = "harvester_ssh_key"
...
rv = {"resource":{resource_type:{resource_name: spec}}}

if convert:
    return ResourceContext(resource_type, resource_name, self.convert_to_hcl(rv))
return rv
```
Same for other resources.

**Review by albinsun — APPROVED** [review]
In general LGTM, with minor comment.

File: harvester_e2e_tests/fixtures/terraform.py (hunk 1 above)
Comment (lanfon72) [reply harvester_e2e_tests/fixtures/terraform.py:148]: https://github.com/harvester/tests/pull/979

**Pull request merged**